RAGchain/preprocess/loader/hwp_table_loader.py

```diff
@@ -0,0 +1,152 @@
+                        else:
+                            result_txt += tag5.get_text()
+                        flag = 0
+                    result_txt += '</td>\n'
+                    line = 0
+        result_txt += '\t </tr>\n'
+    result_txt += '</table>'
+    return result_txt
+
+
+class HwpLoader(BaseLoader):
+    # It only works on Windows
+    def __init__(self, path: str, *args, **kwargs):
+        self.file_path = path
+        self.result = []
+        self.flag = 0
+
+    def convert_hwp_to_hwpx(self):
+        hwp = win32.gencache.EnsureDispatch("HWPFrame.HwpObject")
+        hwp.RegisterModule("FilePathCheckDLL", "FilePathCheckerModule")
+        hwp.HParameterSet.HTableCreation.TableProperties.TreatAsChar = 1
+        hwp.Open(self.file_path)
+        hwp.SaveAs(hwp.Path + "X", "HWPX")
+        hwp.Quit()
+        self.file_path = self.file_path + "x"
+
+    def unzip_hwpx(self, file_path):
+        os.chdir(os.path.dirname(self.file_path))
+        target_path = os.path.join(os.getcwd(), "hwpx")
+        with zipfile.ZipFile(self.file_path, 'r') as zf:
+            zf.extractall(path=target_path)
+        # os.remove(self.file_path)
+
+    def preprocessor(self):
+
+        if self.file_path[-5:] == ".hwpx":
+            self.unzip_hwpx(self.file_path)
+
+        elif self.file_path[-4:] == ".hwp":
+
+            self.convert_hwp_to_hwpx()
+            self.unzip_hwpx(self.file_path)
+
+        else:
+            raise ValueError("The file extension must be .hwp or .hwpx")
+
+        # print("Loading {0}".format(self.file_path))
+
+        pattern = r'</?(?!(?:em|strong)\b)[a-z](?:[^>"\']|"[^"]*"|\'[^\']*\')*>'
+
+        for i, xml in enumerate(spliter(os.path.join(os.getcwd(), "hwpx", "Contents", "section0.xml"))):
+            if i % 2 == 0:
+                self.result.append(re.sub(pattern, '', xml))
+            if i % 2 == 1:
+                self.result.append('<hp:tbl' + xml)
+
+        self.result[0] = self.result[0].strip("""['<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>""")
+
+        for i, xml in enumerate(self.result):
+            if i % 2 == 1:
+                self.result[i] = xml_to_html(xml)
+
+        # print("Preprocessing is done")
+        return self.result
+
+    def load(self) -> List[Document]:
+        document_list = []
+        page = ""
+        if self.flag == 0:
+            self.preprocessor()
+            self.flag = 1
+            return self.load()
+
+        elif self.flag == 1:
+            for i, txt in enumerate(self.result):
+                if i % 2 == 0:
+                    page += txt
+            document_list = [Document(page_content=page, metadata={"source": self.file_path})]
+
+        if os.path.exists(os.path.join(self.file_path, os.pardir, "hwpx")):
+            shutil.rmtree(os.path.join(self.file_path, os.pardir, "hwpx"))
+            os.remove(os.path.join(self.file_path))
+        return document_list
+
+    def load_table(self) -> List[Document]:
```

Comment:
Make load_table() function to private function. 
And return each table to separate documents at load() & lazy_load() functions. 
Please look MathpixMarkdownLoader.
That class split tables to each documents. And at Document object with table contents, it has metadata 'content_type': 'table'. 
I think following that convention might great. Any other idea? Please let me know.